Sources/LanguageServerProtocolJSONRPC/JSONRPCConnection.swift

```diff
@@ -230,6 +230,133 @@ public final class JSONRPCConnection: Connection {
+      case .request:
+        if let id = error.id {
+          // If we know it was a request and we have the request ID, simply reply to the request and tell the client
+          // that we couldn't parse it. That complies with LSP that all requests should eventually get a response.
+          logger.fault(
+            "Replying to request \(id, privacy: .public) with error response because we failed to decode the request"
+          )
+          self.send(.errorResponse(ResponseError(error), id: id))
+          return nil
+        } else {
+          // If we don't know the ID of the request, ignore it and show a notification to the user.
+          // That way the user at least knows that something is going wrong even if the client never gets a response
+          // for the request.
+          logger.fault("Ignoring request because we failed to decode the request and don't have a request ID")
+          sendMessageDecodingErrorNotificationToClient(message: "sourcekit-lsp failed to decode a request")
+          return nil
+        }
+      case .response:
+        if let id = error.id {
+          if let outstanding = self.outstandingRequests.removeValue(forKey: id) {
+            // If we received a response to a request we sent to the client, assume that the client responded with an
+            // error. That complies with LSP that all requests should eventually get a response.
+            logger.fault(
+              "Assuming an error response to request \(id, privacy: .public) because response from client could not be decoded"
+            )
+            outstanding.replyHandler(.failure(ResponseError(error)))
+            return nil
+          } else {
+            // If there's an error in the response but we don't even know about the request, we can ignore it.
+            logger.fault(
+              "Ignoring response to request \(id, privacy: .public) because message could not be decoded but we don't know about the request ID"
+            )
+            return nil
+          }
+        } else {
+          // And if we can't even recover the ID the response is for, we drop it. This means that whichever code in
+          // sourcekit-lsp sent the request will probably never get a reply but there's nothing we can do about that.
+          // Ideally requests sent from sourcekit-lsp to the client would have some kind of timeout anyway.
+          logger.fault("Ignoring response because the ID of the request the response is to could not be recovered")
+          return nil
+        }
+      case .notification:
+        if error.code == .methodNotFound {
+          // If we receive a notification we don't know about, this might be a client sending a new LSP notification
+          // that we don't know about. It can't be very critical so we ignore it without bothering the user with an
+          // error notification.
+          logger.fault("silently ignoring notification because we don't know about it's method")
```

Comment:
```suggestion
          logger.fault("silently ignoring notification as the method is unknown")
```

And a general nitpick: we swap between uppercase and lowercase for the start of the message + this is the only one that says silent, despite a bunch of others being silent